### DISCUSSION TITLE
How to make use of EXT_mesh_gpu_instancing?

### DISCUSSION THREAD

**Discussion by brandon-lockaby** [donmccurdy/glTF-Transform#711] [Q&A] [answered]
I am trying to make sense of how one can use gpu instances in GLTF.

I'm able to get gltf-transform to recognize when instances should be used, and it produces a GLB which uses the EXT_mesh_gpu_instancing extension, but then the instances no longer show up in any of the popular viewers, or when importing into blender.

Sure, you can iterate through the data and create GPU instances in code.  But I was hoping that gpu instancing in gltf would mean it worked in gltf-viewer, gltf.report, modelviewer.dev, blender import, etc, otherwise it's not clear what the functional difference is between dedup and instance.  I am hoping the difference is that you can actually get the viewers to use gpu instanced meshes.

**Comment by donmccurdy** [comment, marked as answer]
Hi @brandon-lockaby! The EXT-prefixed extensions are not published by the Khronos Group (authors and maintainers of the glTF specification), and are not as widely supported. If you want your model to work in as many tools as possible, you'll probably want to use only KHR-prefixed extensions. Unofficial extensions (EXT_ and others) are more commonly used by developers who are trying to optimize their work for a specific project, and know their tooling supports the extensions they're using.

That said — babylon.js (the engine used in https://sandbox.babylonjs.com/) already supports `EXT_mesh_gpu_instancing`. The next three.js release at the end of this month will add `EXT_mesh_gpu_instancing` support as well, and three.js-based viewers (the others in your list) should start to update and gain support in the next few months after that.
